pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,279 @@
+
+    When the number of angles is even (:math:`d` is odd), the QSVT circuit is defined as:
+
+    .. math::
+
+        U_{QSVT} = \Pi_{\phi_1}U\left[\prod^{(d-1)/2}_{k=1}\Pi_{\phi_{2k}}U^\dagger
+        \Pi_{\phi_{2k+1}}U\right]\Pi_{\phi_{d+1}},
+
+
+    and when the number of angles is odd (:math:`d` is even):
+
+    .. math::
+
+        U_{QSVT} = \left[\prod^{d/2}_{k=1}\Pi_{\phi_{2k-1}}U^\dagger\Pi_{\phi_{2k}}U\right]
+        \Pi_{\phi_{d+1}}.
+
+    Here, :math:`U` denotes a block encoding of :math:`A` via :class:`~.BlockEncode` and
+    :math:`\Pi_\phi` denotes a projector-controlled phase shift with angle :math:`\phi`
+    via :class:`~.PCPhase`.
+
+    This circuit applies a polynomial transformation (:math:`Poly^{SV}`) to the singular values of
+    the block encoded matrix:
+
+    .. math::
+
+        \begin{align}
+             U_{QSVT}(A, \phi) &=
+             \begin{bmatrix}
+                Poly^{SV}(A) & \cdot \\
+                \cdot & \cdot
+            \end{bmatrix}.
+        \end{align}
+
+    .. seealso::
+
+        :class:`~.QSVT` and `A Grand Unification of Quantum Algorithms <https://arxiv.org/pdf/2105.02859.pdf>`_
+
+    Args:
+        A (tensor_like): the general :math:`(n \times m)` matrix to be encoded
+        angles (tensor_like): a list of angles by which to shift to obtain the desired polynomial
+        wires (Iterable): the wires the template acts on
+        convention (string): can be set to "Wx" to convert quantum signal processing angles in the
+            "Wx" convention to QSVT angles.
+
+    **Example**
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> A = np.array([[0.1, 0.2], [0.3, 0.4]])
+    >>> angles = np.array([0.1, 0.2, 0.3])
+    >>> @qml.qnode(dev)
+    ... def example_circuit(A):
+    ...     qml.qsvt(A, angles, wires=[0, 1])
+    ...     return qml.expval(qml.PauliZ(wires=0))
+    >>> print(qml.draw(example_circuit)(A))
+    0: ─╭QSVT─┤  <Z>
+    1: ─╰QSVT─┤
+    >>> print(qml.draw(example_circuit, expansion_strategy="device")(A))
+    0: ─╭∏_ϕ(0.10)─╭BlockEncode(M0)─╭∏_ϕ(0.20)─╭BlockEncode(M0)†─╭∏_ϕ(0.30)─┤  <Z>
+    1: ─╰∏_ϕ(0.10)─╰BlockEncode(M0)─╰∏_ϕ(0.20)─╰BlockEncode(M0)†─╰∏_ϕ(0.30)─┤
+    """
+    if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
+        A = qml.math.reshape(A, [1, 1])
+
+    c, r = qml.math.shape(A)
+
+    UA = BlockEncode(A, wires=wires, do_queue=False)
+    projectors = []
+
+    if convention == "Wx":
+        angles = _qsp_to_qsvt(angles)
+        global_phase = (len(angles) - 1) % 4
+
+        if global_phase:
+            qml.exp(qml.Identity(wires=wires), 0.5j * np.pi * (4 - global_phase))
+
+    for idx, phi in enumerate(angles):
+        dim = c if idx % 2 else r
+        projectors.append(PCPhase(phi, dim=dim, wires=wires, do_queue=False))
+
+    return QSVT(UA, projectors, wires=wires)
+
+
+class QSVT(Operation):
+    r"""Implements the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit.
+
```

Comment:
Maybe good form to say from the start that this allows users to specify circuits for block encodings, as required for hardware. Then link saying that the related operation qsvt uses a canonical block encoding, specially tailored for use in simulators